src/ScottPlot/plottables/PlottableScatterHighlight.cs

```diff
@@ -0,0 +1,118 @@
+using System.Drawing;
+using ScottPlot.Config;
+using ScottPlot.Drawing;
+using System.Diagnostics;
+using ScottPlot.plottables;
+
+namespace ScottPlot
+{
+    public class PlottableScatterHighlight : PlottableScatter, IExportable, IHighlightable
+    {
+        protected List<int> highlightedIndexes;
+        public MarkerShape highlightedShape;
+        public float highlightedMarkerSize;
+        public Color highlightedColor;
+        private Pen penLineError;
+
+        public PlottableScatterHighlight(double[] xs, double[] ys, Color color, double lineWidth, double markerSize, string label,
+            double[] errorX, double[] errorY, double errorLineWidth, double errorCapSize, bool stepDisplay, MarkerShape markerShape, LineStyle lineStyle, MarkerShape highlightedShape, Color highlightedColor, double highlightedMarkerSize)
+            : base(xs, ys, color, lineWidth, markerSize, label, errorX,  errorY, errorLineWidth, errorCapSize, stepDisplay, markerShape, lineStyle)
+        {
+            this.highlightedColor = highlightedColor;
+            this.highlightedMarkerSize = (float)highlightedMarkerSize;
+            this.highlightedShape = highlightedShape;
+            this.highlightedIndexes = new List<int>();
+        }
+
+        protected override void DrawPoint(Settings settings, List<PointF> points, int i)
+        {
+            if (highlightedIndexes.BinarySearch(i) >= 0) //Always returns negative number for item not found: https://docs.microsoft.com/en-us/dotnet/api/system.collections.generic.list-1.binarysearch?view=netcore-3.1#System_Collections_Generic_List_1_BinarySearch__0_
+            {
+                MarkerTools.DrawMarker(settings.gfxData, points[i], highlightedShape, highlightedMarkerSize, highlightedColor);
+            }
+            else
+            {
+                base.DrawPoint(settings, points, i);
+            }
+        }
+
+        public void HighlightClear()
+        {
+            highlightedIndexes.Clear();
+        }
+
+        public void HighlightPoint(int index)
+        {
+            int insertPosition = highlightedIndexes.BinarySearch(index);
+            if (insertPosition < 0) {
+                insertPosition = ~insertPosition;   //This might look like witchcraft
+                                                    //However, List<T>.BinarySearch returns the ones-compliment (bitwise inverse) of the index of the next-higher value in the list
+                                                    //if the value is not part of the list.
+                                                    //If there is no higher value it returns the ones-compliment of this.Count()
+                                                    //Source: https://docs.microsoft.com/en-us/dotnet/api/system.collections.generic.list-1.binarysearch?view=netcore-3.1#System_Collections_Generic_List_1_BinarySearch__0_
+            }
+            highlightedIndexes.Insert(insertPosition, index);
+        }
```

Comment:
I'm a little confused here. Not about the bit-inverse binary search result (which is cool! I didn't know that), but about what the intent of this function is. 

Currently if you do this:

```cs
highlightPlot.HighlightPointNearest(8);
highlightPlot.HighlightPointNearest(8);
highlightPlot.HighlightPointNearest(8);
highlightPlot.HighlightPointNearest(8);
highlightPlot.HighlightPointNearest(8);
```

It will keep adding the same index to the `highlightedIndexes` list. Is this the intended behavior? 

Perhaps this method could be simplified to:

```cs
public void HighlightPoint(int index)
{
    int searchResult = highlightedIndexes.BinarySearch(index);

    if (searchResult < 0)
        highlightedIndexes.Insert(~searchResult, index);
}
```

or even simpler:

```cs
public void HighlightPoint(int index)
{
    if (highlightedIndexes.Contains(index))
        highlightedIndexes.Add(index);
}
```

Reply:
Is the motivation that BinarySearch is faster on ordered lists, and it's important because it's called frequently (in the render loop)?

If performance is the concern, maybe a `bool[] isHighlighted` the same length as `double[] xs` is an alternative way to keep track of this. Then the render loop would require no searching at all. It's a little more memory, but not much.